java/ql/src/semmle/code/java/dataflow/internal/ContainerFlow.qll

```diff
@@ -88,6 +91,290 @@ class ContainerType extends RefType {
+        "java.util;NavigableSet;true;lower;(Object);;Element of Argument[-1];ReturnValue;value",
+        "java.util;NavigableSet;true;pollFirst;();;Element of Argument[-1];ReturnValue;value",
+        "java.util;NavigableSet;true;pollLast;();;Element of Argument[-1];ReturnValue;value",
+        "java.util;NavigableSet;true;subSet;(Object,boolean,Object,boolean);;Element of Argument[-1];Element of ReturnValue;value",
+        "java.util;NavigableSet;true;tailSet;(Object,boolean);;Element of Argument[-1];Element of ReturnValue;value",
+        "java.util;Scanner;true;next;(Pattern);;Argument[-1];ReturnValue;taint",
+        "java.util;Scanner;true;next;(String);;Argument[-1];ReturnValue;taint",
+        "java.util;SortedMap;true;headMap;(Object);;MapKey of Argument[-1];MapKey of ReturnValue;value",
+        "java.util;SortedMap;true;headMap;(Object);;MapValue of Argument[-1];MapValue of ReturnValue;value",
+        "java.util;SortedMap;true;subMap;(Object,Object);;MapKey of Argument[-1];MapKey of ReturnValue;value",
+        "java.util;SortedMap;true;subMap;(Object,Object);;MapValue of Argument[-1];MapValue of ReturnValue;value",
+        "java.util;SortedMap;true;tailMap;(Object);;MapKey of Argument[-1];MapKey of ReturnValue;value",
+        "java.util;SortedMap;true;tailMap;(Object);;MapValue of Argument[-1];MapValue of ReturnValue;value",
+        "java.util;SortedSet;true;first;();;Element of Argument[-1];ReturnValue;value",
+        "java.util;SortedSet;true;headSet;(Object);;Element of Argument[-1];Element of ReturnValue;value",
+        "java.util;SortedSet;true;last;();;Element of Argument[-1];ReturnValue;value",
+        "java.util;SortedSet;true;subSet;(Object,Object);;Element of Argument[-1];Element of ReturnValue;value",
+        "java.util;SortedSet;true;tailSet;(Object);;Element of Argument[-1];Element of ReturnValue;value",
+        "java.util.concurrent;TransferQueue;true;tryTransfer;(Object,long,TimeUnit);;Argument[0];Element of Argument[-1];value",
+        "java.util.concurrent;TransferQueue;true;transfer;(Object);;Argument[0];Element of Argument[-1];value",
+        "java.util.concurrent;TransferQueue;true;tryTransfer;(Object);;Argument[0];Element of Argument[-1];value",
+        "java.util.stream;Node;true;copyInto;(Object[],int);;Element of Argument[-1];ArrayElement of Argument[0];value",
+        "java.util.stream;Node<>$OfPrimitive;true;copyInto;(Object,int);;Element of Argument[-1];ArrayElement of Argument[0];value",
+        "java.util.stream;SpinedBuffer;true;copyInto;(Object[],int);;Element of Argument[-1];ArrayElement of Argument[0];value",
+        "java.util.stream;SpinedBuffer<>$OfPrimitive;true;copyInto;(Object,int);;Element of Argument[-1];ArrayElement of Argument[0];value",
+        "java.util;List;false;copyOf;(Collection);;Element of Argument[0];Element of ReturnValue;value",
+        "java.util;List;false;of;(Object[]);;Element of Argument[0];Element of ReturnValue;value",
```

Comment:
OOI, what does the empty `of` in `;of;` mean?

Reply:
That's just the name of the varargs method [`List::of`](https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/util/List.html#of(E...)) and not related to the DSL.

Reply:
Ah, good.